src/databricks/labs/ucx/source_code/python_libraries.py

```diff
@@ -71,12 +75,25 @@ def _install_library(self, path_lookup: PathLookup, library: Path) -> MaybeDepen
         # invoke pip install via subprocess to install this library into self._lib_install_folder
         venv = self._temporary_virtual_environment(path_lookup).as_posix()
         existing_packages = os.listdir(venv)
-        pip_install_arguments = ["pip", "install", library.name, "-t", venv]
-        try:
-            subprocess.run(pip_install_arguments, check=True)
-        except CalledProcessError as e:
-            problem = DependencyProblem("library-install-failed", f"Failed to install {library}: {e}")
-            return MaybeDependency(None, [problem])
+        if library.suffix == ".egg":
+            # Setup requires the install directory to be in the python path
+            python_path = os.environ.get("PYTHONPATH", "")
+            if venv not in python_path:
+                os.environ["PYTHONPATH"] = f"{python_path}:{venv}"
+            setup(
+                script_args=["-q", "easy_install", "-v", "--no-deps", "--always-unzip", "-d", venv, library.as_posix()],
```

Comment:
After some research, I learned the `easy_install` script can be invoked as follows. However, there are some downsides to this approach. I had to add the `--no-deps` flag as it tried to install itself as a dependency from PyPI, which failed.